Reorg valid config schema for regtest

shahzebsiddiqui opened the pull request:
Create directory examples/config_schemas/valid and store all valid configuration here.

Remove test so we have one test to validate all yaml files. Its sufficient to run ``validate``
against all examples since we expect this to pass. If it fails it will break on the first validate
stage and test will be marked fail.
Change logic on file extension when loading schema. Ensure that .yml, .yaml, and .json are searched
otherwise any arbitrary file can be loaded with extension (ml, or json) without a period.

Fix #254

shahzebsiddiqui committed:
Create directory examples/config_schemas/valid and store all valid configuration here.
Remove test so we have one test to validate all yaml files. Its sufficient to run ``validate``
against all examples since we expect this to pass. If it fails it will break on the first validate
stage and test will be marked fail.
Change logic on file extension when loading schema. Ensure that .yml, .yaml, and .json are searched
otherwise any arbitrary file can be loaded with extension (ml, or json) without a period.

vsoch commented on buildtest/buildsystem/schemas/utils.py:
```diff
@@ -25,9 +25,9 @@ def load_schema(path):
+        if path.endswith(".json"):
             schema = json.loads(fd.read())
-        elif path.endswith("ml"):
+        elif path.endswith(".yaml") or path.endswith(".yml"):
```
Nit: this would be more elegant with:

```python
if re.search("(yaml|yaml)^", path):
```
Generally when you have to do more than one if this, if that, it's good to try and look for a regular expression that can represent both.

shahzebsiddiqui replied:
thats good suggestion. Regex are very powerful and this would be good example. I need to get more comfortable with regex, i have some understanding but practically haven't used it much in the codebase as you can see.

vsoch commented on tests/menu/test_config.py:
```diff
@@ -21,26 +22,10 @@ def test_config_restore():
-    example_schema = os.path.join(
-        pytest_root, "config_schema_examples", "local-example.yml"
-    )
+    valid_schema_dir = os.path.join(pytest_root,"examples","config_schemas","valid")
```
Does this validate with black?

shahzebsiddiqui replied:
i ran black against all files, seems like line 27 ``valid_schema_dir`` didn't change so i guess it is valid with black.

shahzebsiddiqui committed:
Add local-example.json which is a json representation of local-example.yml
Fix path issue reported by Travis build
use re.search to check for .json, .yml and .yaml extension when loading configuration
apply black to code

vsoch commented on buildtest/buildsystem/schemas/utils.py:
```diff
@@ -25,9 +25,9 @@ def load_schema(path):
             schema = json.loads(fd.read())
-        elif path.endswith("ml"):
+        elif re.search("(.yaml|.yml)$", path):
             schema = yaml.load(fd.read(), Loader=yaml.SafeLoader)
```
This isn't matching what you think - a period means "match anything" For example:

```python
re.search("(.yaml|.yml)$", 'ayaml')                                     
<re.Match object; span=(0, 5), match='ayaml'>
```
What you want is an explicit period, and since it's shared between the two, put it outside of the or:

```python
re.search("[.](yaml|yml)$", 'ayaml')                                    
# No match (correct)
re.search("[.](yaml|yml)$", '.yaml')                                    
<re.Match object; span=(0, 5), match='.yaml'>
re.search("[.](yaml|yml)$", 'config.yaml')                              
<re.Match object; span=(6, 11), match='.yaml'>
```
The same is true for the json above - it needs to be
```python
 re.search("[.]json$", path):
```

shahzebsiddiqui replied:
ahh that's good catch! i totally forgot about the period since it's special character in regex. I will fix this.

vsoch reviewed (changes requested):
Almost! One small bug with the regular expressions.

shahzebsiddiqui committed:
fix regular expression search for file extension for configuration file

Signed-off-by: Shahzeb Siddiqui <shahzebmsiddiqui@gmail.com>
